nexus/src/app/instance.rs

```diff
@@ -221,54 +251,46 @@ impl super::Nexus {
             .map_err(|e| Error::internal_error(&format!("{:#}", &e)))
             .internal_context("looking up output from instance create saga")?;
 
-        // TODO-correctness TODO-robustness TODO-design It's not quite correct
-        // to take this instance id and look it up again.  It's possible that
-        // it's been modified or even deleted since the saga executed.  In that
-        // case, we might return a different state of the Instance than the one
-        // that the user created or even fail with a 404!  Both of those are
-        // wrong behavior -- we should be returning the very instance that the
-        // user created.
-        //
-        // How can we fix this?  Right now we have internal representations like
-        // Instance and analaogous end-user-facing representations like
-        // Instance.  The former is not even serializable.  The saga
-        // _could_ emit the View version, but that's not great for two (related)
-        // reasons: (1) other sagas might want to provision instances and get
-        // back the internal representation to do other things with the
-        // newly-created instance, and (2) even within a saga, it would be
-        // useful to pass a single Instance representation along the saga,
-        // but they probably would want the internal representation, not the
-        // view.
-        //
-        // The saga could emit an Instance directly.  Today, Instance
-        // etc. aren't supposed to even be serializable -- we wanted to be able
-        // to have other datastore state there if needed.  We could have a third
-        // InstanceInternalView...but that's starting to feel pedantic.  We
-        // could just make Instance serializable, store that, and call it a
-        // day.  Does it matter that we might have many copies of the same
-        // objects in memory?
-        //
-        // If we make these serializable, it would be nice if we could leverage
-        // the type system to ensure that we never accidentally send them out a
-        // dropshot endpoint.  (On the other hand, maybe we _do_ want to do
-        // that, for internal interfaces!  Can we do this on a
-        // per-dropshot-server-basis?)
+        // If the caller asked to start the instance, kick off that saga.
+        // There's a window in which the instance is stopped and can be deleted,
+        // so this is not guaranteed to succeed, and its result should not
+        // affect the result of the attempt to create the instance.
```

Comment:
It's not totally infallible; you could still fail to create the VM. But otherwise you're right: in the old code, if you said `start: true` and the VM failed to start, the whole saga would unwind and the created VM would be destroyed. Now the behavior matches what you'd get if you passed `start: false` and then issued a separate `instance start` command once the VM was successfully created. I happen to like that symmetry, but don't feel especially strongly about it.

If (or when) we want the old behavior back, we can get it by trying to turn the start saga into a subsaga of the create saga. I don't think that's impossible, but there's a mild impedance mismatch in there that we'd have to resolve: the start saga expects to get a `db::Instance` as a parameter; to use it as a subsaga, the create saga's `make_saga_dag` has to be able to produce the new instance record in advance; I think that might be possible in principle, but we'd have to rework the create saga to do that, since it currently creates the instance record inline.